### PR TITLE
features: Manually setting the validation set for multi-output task

### PR DESCRIPTION
**Pull request by lizhuoq**
## Why are these changes needed?

For original multi-output tasks where the eval_method is holdout, manual setting of the validation set was not possible. This commit introduces a new feature allowing manual setting of the validation set for multi-output tasks.
```python
model = MultiOutputRegressor(
    AutoML(
        task="regression",
        time_budget=1,
        eval_method="holdout",
        multioutput_train_size=len(X_train)
    )
)
model.fit(
    pd.concat([X_train, X_val]),
    pd.concat([y_train, y_val])
)
```
## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by lizhuoq** [comment]
@microsoft-github-policy-service agree

**Review by Programmer-RD-AI — APPROVED** [review]
LGTM!
